### PR TITLE
[Behat][Promotion] Percentage discount on products from specific taxon

### PR DESCRIPTION
**Pull request by Zales0123**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | https://github.com/Sylius/Sylius/issues/3368
| License       | MIT
| Doc PR        |

This is next portion of scenarios about promotions - however this time it would finally require some more coding :smile: However, for now there is only feature, which describes new promotions case - applying promotions only to product from specific taxon.

### CODE REVIEW

File: features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    Background:
+        Given the store operates on a single channel in "France"
+        And store classifies its products as "T-Shirts", "Mugs" and "Books"
+        And the store has a product "PHP T-Shirt" priced at "€100.00"
+        And it belongs to "T-Shirts"
+        And the store has a product "PHP Mug" priced at "€20.00"
+        And it belongs to "Mugs"
+        And there is a promotion "T-Shirts promotion"
+        And it gives "20%" percentage discount to every order item containing products from "T-Shirts" taxon
```
Comment (michalmarcinkowski) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: IMO: to every order item ~~containing products~~ from "T-Shirts" taxon

File: features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    @todo
+    Scenario: Receiving percentage discount on t-shirt
```
Comment (michalmarcinkowski) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: on **a T-S**hirt?

File: features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        When I add product "PHP T-Shirt" to the cart
+        Then it should be discounted at "-€20.00"
```
Comment (michalmarcinkowski) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: `its price should be decreased by "€20.00"`?

File: features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        And my cart total should be "€80.00"
+        And my discount should be "-€20.00"
+
+    @todo
+    Scenario: Receiving percentage discount on multiple t-shirts
```
Comment (michalmarcinkowski) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: ...on a multiple items from promoted taxon

File: features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        When I add 3 products "PHP T-Shirt" to the cart
+        Then they should be discounted at "-€60.00"
+        And my cart total should be "€240.00"
+        And my discount should be "-€60.00"
+
+    @todo
+    Scenario: Receiving percentage discount does not affect other categories
```
Comment (michalmarcinkowski) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: Receiving percentage discount only on items from promoted taxon

File: features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        And product "PHP Mug" should not be discounted
+        And my cart total should be "€100.00"
+        And my discount should be "-€20.00"
+
+    @todo
+    Scenario: Receiving different discounts on different categories
+        Given the store has a product "PHP Book" priced at "€50.00"
+        And it belongs to "Books"
+        And there is a promotion "Books promotion"
+        And it gives "50%" percentage discount to every order item containing products from "Books" taxon
```
Comment (michalmarcinkowski) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: Why creating another product? Can't we simply create a promotion for Mugs and remove the first two steps?

File: features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        And I add product "PHP Mug" to the cart
+        Then product "PHP T-Shirt" should be discounted at "-€20.00"
+        And product "PHP Mug" should not be discounted
+        And my cart total should be "€100.00"
+        And my discount should be "-€20.00"
+
+    @todo
+    Scenario: Receiving different discounts on different categories
+        Given the store has a product "PHP Book" priced at "€50.00"
+        And it belongs to "Books"
```
Comment (michalmarcinkowski) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: Why creating another product? Can't we simply create a promotion for Mugs and remove these two steps?

Comment (Zales0123) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: :+1:

File: features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        When I add product "PHP T-Shirt" to the cart
+        And I add product "PHP Mug" to the cart
+        Then product "PHP T-Shirt" price should be decreased by "€20.00"
+        And product "PHP Mug" price should not be decreased
+        And my cart total should be "€100.00"
+        And my discount should be "-€20.00"
+
+    @todo
+    Scenario: Receiving different discounts on different categories
+        And there is a promotion "Mugs promotion"
```
Comment (michalmarcinkowski) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: Given?

File: features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+    Background:
+        Given the store operates on a single channel in "France"
+        And store classifies its products as "T-Shirts" and "Mugs"
+        And the store has a product "PHP T-Shirt" priced at "€100.00"
+        And it belongs to "T-Shirts"
+        And the store has a product "PHP Mug" priced at "€20.00"
+        And it belongs to "Mugs"
+        And there is a promotion "T-Shirts promotion"
+        And it gives "20%" percentage discount to every order item from "T-Shirts" taxon
```
Comment (pjedrzejewski) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: 20% percentage??? This step sounds almost technical.

``And it gives 20% off every product classified as "T-Shirts"``

WDYT?

Comment (pjedrzejewski) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: Or we can replace ``off`` with ``discount``, does not matter imho.

Comment (Zales0123) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: It has been done this way in previous scenarios ;) but sure, I think we can skip this ``percentage`` word.

File: features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+    @todo
+    Scenario: Receiving percentage discount on a single item from promoted taxon
```
Comment (pjedrzejewski) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: ``promoted taxon`` does not sound right.

Receiving percentage discount on a single item from specific taxon

To be honest I am not sure how to associate product with taxon:

* under taxon
* from taxon
* classified as taxon

...

File: features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    Scenario: Receiving percentage discount only on items from promoted taxon
+        When I add product "PHP T-Shirt" to the cart
+        And I add product "PHP Mug" to the cart
+        Then product "PHP T-Shirt" price should be decreased by "€20.00"
+        And product "PHP Mug" price should not be decreased
+        And my cart total should be "€100.00"
+        And my discount should be "-€20.00"
+
+    @todo
+    Scenario: Receiving different discounts on different categories
```
Comment (pjedrzejewski) [inline features/promotion/receiving_percentage_discount_on_products_from_specific_taxon.feature]: We do not really talk about categories, we should avoid that.

**Comment by michalmarcinkowski** [comment]
:+1:

**Pull request merged**